### PR TITLE
Clean up documentation for example code.

### LINKED ISSUE #380: Threaded Example: Cleaning up top-level documentation.
- [x] Clean up top-level documentation for explaining threaded example communication/authentication requirements and how the example does or does not satisfy them.
- [x] Add doc strings to all functions and types.
- [x] Ensure existing doc strings are correct and thorough.

### PR DESCRIPTION
**Pull request by gatoWololo**
Closes #380

### CODE REVIEW

File: examples/threaded_example/threaded.rs
Comment (marsella) [inline examples/threaded_example/threaded.rs:1]: issue: I think the top-level documentation's Trust Model section (or maybe documentation on the `Coordinator`? whichever you think is more likely to be read by someone looking at the example) should note that a coordinator is not _necessary_ to run the library. Something like,

> A trusted `Coordinator` is not required to run threshold ECDSA protocol. Networking models with less trust include authenticated point-to-point channels between the workers or an untrusted relay node that routes authenticated messages between the workers.

**Review by marsella — CHANGES_REQUESTED** [review]
This is a great improvement! The updated type names throughout are much better and make the whole example easier to understand, in my opinion.

File: examples/threaded_example/threaded.rs [hunk 1]
```diff
@@ -167,14 +177,14 @@ fn main() -> anyhow::Result<()> {
...
-    do_tss_ecdsa(runner, cli.protocol_executions)?;
+    let mut coordinator = Coordinator::new(worker_messages, workers_rx, num_workers);
+    coordinator.tss_ecdsa()?;
 
     Ok(())
 }
 
-/// Helper type for running sub-protocols for the coordinator.
-struct SubProtocolRunner {
+/// Coordinator responsible for running sub-protocols.
```
[the remark this answers is not in the document]
Comment (gatoWololo) [reply examples/threaded_example/threaded.rs:186]: That totally makes sense!

**Review by marsella — APPROVED** [review]
Looks good, thanks for addressing!

**Pull request merged**